plugin.video.fullycharged/plugintools.py

```diff
@@ -0,0 +1,582 @@
+        dev = dev.encode("utf-8")
+    except:
+        pass
+
+    _log("get_localized_string ->'"+dev+"'")
+
+    return dev
+
+def keyboard_input(default_text="", title="", hidden=False):
+    _log("keyboard_input default_text='"+default_text+"'")
+
+    keyboard = xbmc.Keyboard(default_text,title,hidden)
+    keyboard.doModal()
+    
+    if (keyboard.isConfirmed()):
+        tecleado = keyboard.getText()
+    else:
+        tecleado = ""
+
+    _log("keyboard_input ->'"+tecleado+"'")
+
+    return tecleado
+
+def message(text1, text2="", text3=""):
+    _log("message text1='"+text1+"', text2='"+text2+"', text3='"+text3+"'")
+
+    if text3=="":
+        xbmcgui.Dialog().ok( text1 , text2 )
+    elif text2=="":
+        xbmcgui.Dialog().ok( "" , text1 )
+    else:
+        xbmcgui.Dialog().ok( text1 , text2 , text3 )
+
+def message_yes_no(text1, text2="", text3=""):
+    _log("message_yes_no text1='"+text1+"', text2='"+text2+"', text3='"+text3+"'")
+
+    if text3=="":
+        yes_pressed = xbmcgui.Dialog().yesno( text1 , text2 )
+    elif text2=="":
+        yes_pressed = xbmcgui.Dialog().yesno( "" , text1 )
+    else:
+        yes_pressed = xbmcgui.Dialog().yesno( text1 , text2 , text3 )
+
+    return yes_pressed
+
+def selector(option_list,title="Select one"):
+    _log("selector title='"+title+"', options="+repr(option_list))
+
+    dia = xbmcgui.Dialog()
+    selection = dia.select(title,option_list)
+
+    return selection
+
+def set_view(view_mode, view_code=0):
+    _log("set_view view_mode='"+view_mode+"', view_code="+str(view_code))
+
+    # Set the content for extended library views if needed
+    if view_mode==MOVIES:
+        _log("set_view content is movies")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"movies" )
+    elif view_mode==TV_SHOWS:
+        _log("set_view content is tvshows")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"tvshows" )
+    elif view_mode==SEASONS:
+        _log("set_view content is seasons")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"seasons" )
+    elif view_mode==EPISODES:
+        _log("set_view content is episodes")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"episodes" )
+
+    # Reads skin name
+    skin_name = xbmc.getSkinDir()
+    _log("set_view skin_name='"+skin_name+"'")
+
+    try:
+        if view_code==0:
+            _log("set_view view mode is "+view_mode)
+            view_codes = ALL_VIEW_CODES.get(view_mode)
+            view_code = view_codes.get(skin_name)
+            _log("set_view view code for "+view_mode+" in "+skin_name+" is "+str(view_code))
+            xbmc.executebuiltin("Container.SetViewMode("+str(view_code)+")")
+        else:
+            _log("set_view view code forced to "+str(view_code))
+            xbmc.executebuiltin("Container.SetViewMode("+str(view_code)+")")
+    except:
+        _log("Unable to find view code for view mode "+str(view_mode)+" and skin "+skin_name)
+
+f = open( os.path.join( os.path.dirname(__file__) , "addon.xml") )
```

Comment:
This is a strange way to get the addon ID, nevermind that "open with" is the correct way to do the file read.
Is there something I'm missing here like the addon is trying to read a different addon's settings?